test/clj_kondo/main_test.clj

```diff
@@ -2529,6 +2529,36 @@ foo/foo ;; this does use the private var
 
 " {:linters {:unsorted-required-namespaces {:level :warning}}})))))
 
+(deftest require-prefix-form-containing-periods-test
+  (is (= '({:col     32
+            :file    "<stdin>"
+            :level   :error
+            :message "Lib names inside prefix lists must not contain periods."
+            :row     4})
+         (lint! "(ns baz
+                   (:require [clj-kondo.impl.analyzer
+                              [babashka :as baz]
+                              [foo.bar :as quux]]))"
+                {:linters {:unused-namespace {:level :off}}})))
+  (is (= '({:col     32
+            :file    "<stdin>"
+            :level   :error
+            :message "Lib names inside prefix lists must not contain periods."
+            :row     3})
+         (lint! "(ns baz
+                   (:require [clj-kondo.impl.analyzer
+                              [foo.bar :as baz]]))"
+                {:linters {:unused-namespace {:level :off}}})))
+  (is (= '({:col     31
+            :file    "<stdin>"
+            :level   :error
+            :message "Lib names inside prefix lists must not contain periods."
+            :row     3})
+         (lint! "(ns baz
+                   (:require [clj-kondo.impl.analyzer
+                              foo.bar]))"
+                {:linters {:unused-namespace {:level :off}}}))))
+
```

Comment:
Can you explain what the difference in test is between the first and second one? Maybe clarify using `testing`.

Reply:
The first one contains one valid lib name (`babashka`), whereas the second one does not. On second thought though, it doesn't seem like a valuable test case. I think I'll get rid of it, and replace it with a test case that contains multiple invalid lib names.

Reply:
This is changed.